driver-core/src/main/com/mongodb/internal/connection/ScramShaAuthenticator.java

```diff
@@ -72,6 +77,13 @@ public String getMechanismName() {
         return authMechanism.getMechanismName();
     }
 
+    @Override
```

Comment:
You can DRY this up like this:

     protected void appendSaslStartOptions(final BsonDocument saslStartCommand) {
         saslStartCommand. append("options", new BsonDocument("skipEmptyExchange", new BsonBoolean(true)))
     }

and make the one in the base class do nothin.

Reply:
I have added this.